pennylane/operation.py

```diff
@@ -763,6 +770,51 @@ class DiagonalOperation(Operation):
     """
     # pylint: disable=abstract-method
 
+    @classmethod
+    def _eigvals(cls, *params):
+        """Eigenvalues of the operator.
+
+        The order of the eigenvalues needs to match the order of
+        the computational basis vectors.
+
+        This is a *class method* that must be defined for all
+        new diagonal operations, that returns the eigenvalues
+        of the operator in the computational basis.
+
+        This private method allows eigenvalues to be computed
+        directly without instantiating the operators first.
+
+        To return the eigenvalues of *instantiated* operators,
+        please use the :attr:`~.Operator.eigvals` property instead.
+
+        **Example:**
+
+        >>> qml.RZ._eigvals(0.5)
+        >>> array([0.96891242-0.24740396j, 0.96891242+0.24740396j])
+
+        Returns:
+            array: eigenvalue representation
```

Comment:
Looks like it's missing `Args:`

Reply:
I copy-pasted this from the definition of `_matrix`. What is the syntax for documenting varargs?

Reply:
Oh, good point. I'm not 100% sure. Elsewhere in the codebase, we just do

```
Args:
    params (tuple[allowed types]): details
```

for `def func(*params)`. Not sure if this is the best approach, but something I can look into and standardize.